master/internal/api_tasks.go

```diff
@@ -247,14 +249,72 @@ func (a *apiServer) GetTaskAcceleratorData(
 	ctx context.Context,
 	req *apiv1.GetTaskAcceleratorDataRequest,
 ) (*apiv1.GetTaskAcceleratorDataResponse, error) {
-	return nil, grpcutil.UnimplementedError
+	if req.TaskId == "" {
+		return nil, status.Error(codes.InvalidArgument, "task ID missing")
+	}
+	var accelerationData []*apiv1.AcceleratorData
+	rows, err := db.Bun().NewSelect().
+		ColumnExpr("a1.*").
+		TableExpr("allocation_accelerators a1").
+		Join("LEFT JOIN allocations a2 ON a1.allocation_id = a2.allocation_id").
+		Where("a2.task_id = ?", req.TaskId).
+		Rows(ctx)
```

Comment:
1. In ```/pkg/model``` I think we should define a ```type AcceleratorData struct``` then pass this around in ```AddAllocationAcceleratorData/SetAcceleratorData```

2. We should make it a bun type 

not necessarly the ORM features just has the "bun" tags for the fields as an example like Project 
https://github.com/determined-ai/determined/blob/262c33ab24c3f5e521adde9bf71a3f889b74ba1c/master/pkg/model/project.go#L13

3. Then we can replace this Rows(ctx) with 

```
res := &model.AcceleratorData{}
...Scan(ctx, res)
for _, r := range res {
    // convert our model object to proto object
}
```

Reply:
Done!